lib/projectile.js

```diff
@@ -20,12 +20,18 @@ Projectile.prototype.draw = function() {
   return this;
 };
 
-Projectile.prototype.move = function(ship) {
+Projectile.prototype.move = function() {
   if (this.y > -600) {
     this.x += 10*Math.sin(this.angle);
     this.y -= 10*Math.cos(this.angle);
   };
   return this;
 }
 
+Projectile.prototype.edgeCheck = function (x, projectileArray) {
+  if (this.x < 0 || this.x > canvas.width || this.y < 0 || this.y > canvas.height) {
+    projectileArray.splice(x, 1)
+  }
+};
+
```

Comment:
nice addition; makes a lot of sense to take the logic of removing the functionality of removing projectiles out of the game_runner class